scipy/optimize/optimize.py

```diff
@@ -776,12 +777,13 @@ def _line_search_wolfe12(f, fprime, xk, pk, gfk, old_fval, old_old_fval,
 
     if ret[0] is None:
         # line search failed: try different one.
-        with warnings.catch_warnings():
-            warnings.simplefilter('ignore', LineSearchWarning)
-            kwargs2 = {}
-            for key in ('c1', 'c2', 'amax'):
-                if key in kwargs:
-                    kwargs2[key] = kwargs[key]
+        kwargs2 = {}
+        for key in ('c1', 'c2', 'amax'):
+            if key in kwargs:
+                kwargs2[key] = kwargs[key]
+        with suppress_warnings() as sup:
+            sup.filter(LineSearchWarning,
+                       "The line search algorithm did not converge")
```

Comment:
You can do this (also in the other files) outside tests. I am personally not sure that it is great outside tests. 

If there is any way to avoid the filter/warning completely, I think that is much better, but I can see its tricky in scipy, in numpy all of the ones we had were solvable easily with `np.errstate` luckily.